src/_pytest/_code/code.py

```diff
@@ -633,13 +633,13 @@ def getrepr(
         )
         return fmt.repr_excinfo(self)
 
-    def match(self, regexp: "Union[str, Pattern]") -> bool:
+    def match(self, regexp: "Union[str, Pattern]") -> "Literal[True]":
         """
-        Check whether the regular expression 'regexp' is found in the string
-        representation of the exception using ``re.search``. If it matches
-        then True is returned (so that it is possible to write
-        ``assert excinfo.match()``). If it doesn't match an AssertionError is
-        raised.
+        Check whether the regular expression `regexp` matches the string
+        representation of the exception using :func:`python:re.search`.
+        If it matches `True` is returned (so that it is possible to write
```

Comment:
Is there a  reason to do `assert excinfo.match()` as opposed to just `excinfo.match()`? If not, I would just omit the comment in the parenthesis (or move it to a code comment) so it doesn't sound like an endorsement.